packages/blockchain-link-utils/src/stellar.ts

```diff
@@ -124,6 +124,36 @@ export const transformTransaction = (
                 nativeAmount = '0';
             }
             break;
+        case 'changeTrust': {
+            // Only support regular assets, not liquidity pool shares
+            if (
+                rawOp.line.getAssetType() !== 'credit_alphanum4' &&
+                rawOp.line.getAssetType() !== 'credit_alphanum12'
+            ) {
+                return baseTx;
+            }
+
+            if (descriptor !== fromAddress) {
+                return baseTx;
+            }
+
+            const line = rawOp.line as Asset;
+            const assetCode = line.getCode();
+            const isRemoval = new BigNumber(rawOp.limit).isZero();
+
+            return {
+                ...baseTx,
+                type: 'self',
+                stellarSpecific: {
+                    ...baseTx.stellarSpecific!,
+                    operationType: 'changeTrust',
+                    changeTrust: {
+                        assetCode,
+                        isRemoval,
```

Comment:
```suggestion
                        limit: rawOp.limit,
```
I guess it would be good to understand it as "limit"

Reply:
In the change trust operation, the limit represents the maximum amount of an asset the account can hold. If it's set to 0, it means the user can't hold the asset at all, so it won't appear in their wallet.

I think keeping `isRemoval` is a great choice. In 99.99% of cases, users really don't care about what the specific limit is; they only care about whether they are adding a trust line or removing one.

Reply:
would it make sense to change operation naming to `addTokenTrust` and `removeTokenTrust`?

Maybe not, I am not stellar expert. However, by this change we would avoid having 1 more attribute 😀

Reply:
I tend to keep `changeTrust` because it is the name used in various Stellar documents.

<img width="2538" height="1127" alt="Screenshot From 2025-11-19 15-27-09" src="https://github.com/user-attachments/assets/9d8cfc6e-5482-41f7-b440-a57c15237eeb" />

https://stellar.github.io/js-stellar-sdk/Operation.html#.changeTrust

Reply:
limit 👀